File: e54/java/Solution.java
import java.util.ArrayList;
public class Solution {
    /**
     * @param str: A string
     * @return An integer
     */
    public int atoi(String str) {
        // write your code here
		if (str == null || str == "") {
			return 0;
		}
		int sign;
		int begin;
		char[] chars = str.trim().toCharArray();
		if (chars[0] == '-') {
			sign = -1;
			begin = 1;
		}
		else if (chars[0] == '+') {
			sign = 1;
			begin = 1;
		}
		else {
			sign = 1;
			begin = 0;
		}
		ArrayList<Integer> ints = new ArrayList<Integer>();
		for(int i=begin; i < chars.length; i++) {
			if (chars[i] >= '0' && chars[i] <= '9') {
				ints.add((int)(chars[i]-'0'));
			}
			else {
				break;
			}
		}
		long result = 0;
		for (int j=0; j<ints.size(); j++) {
			result += ints.get(j) * (Math.pow(10,(ints.size()-1-j)));
		}
		if (result > 2147483647 && sign == 1)
			return 2147483647;
		else if (result > 2147483648L && sign ==  -1)
			return -2147483648;
		else
			return (int)result * sign;
    }
	public static void main(String[] args) {
		System.out.println(new Solution().atoi("  -1234562"));
		System.out.println(new Solution().atoi("145.62"));
		System.out.println(new Solution().atoi(args[0]));
	}
}

File: e54/python/solution.py
import sys
class Solution:
    # @param str: a string
    # @return an integer
    def atoi(self, str):
        # write your code here
        if str is None or str == "":
            return 0
        str = str.strip()
        sign = 1
        if str[0] == '-':
            sign = -1
            str = str[1:]
        elif str[0] == '+':
            str = str[1:]
        else:
            pass

        ints = []
        digits = []
        points = 0
        for c in str:
            if c >= '0' and c <= '9':
                ints.append(c)
            else:
                break
        if ints == []:
            return 0

        ret = 0
        for i in range(len(ints)):
            ret += (ord(ints[i]) - ord('0')) * 10 ** (len(ints)-1-i)
        if ret > 2147483647 and sign == 1:
            return 2147483647
        if ret > 2147483648 and sign == -1:
            return -2147483648

        return sign * ret

print Solution().atoi("  -51235")
print Solution().atoi(sys.argv[1])
